addons/refuel/functions/fnc_makeSource.sqf

```diff
@@ -25,19 +25,22 @@ if !(EGVAR(common,settingsInitFinished)) exitWith {
     EGVAR(common,runAtSettingsInitialized) pushBack [FUNC(makeSource), _this];
 };
 
+if (!GVAR(enabled)) exitWith {};
+
 params [
     ["_source", objNull, [objNull]],
     ["_fuelCargo", 0, [0]],
     ["_hooks", nil, [[]]]
 ];
-TRACE_3("makeSource",_source,_fuelCargo,_hooks);
 
-private _fuelCargoConfig = getNumber (configOf _source >> QGVAR(fuelCargo));
+private _fuelCargoConfig = _source call FUNC(getFuelCargo);
+
+TRACE_4("makeSource",_source,_fuelCargo,_hooks,_fuelCargoConfig);
 
 if (
     isNull _source
     || {_fuelCargo < 0 && {!(_fuelCargo in [REFUEL_INFINITE_FUEL, REFUEL_DISABLED_FUEL])}}
-    || {_fuelCargo != 0 && {_fuelCargo == _fuelCargoConfig}}
+    || {_fuelCargo == REFUEL_DISABLED_FUEL && {_fuelCargoConfig == REFUEL_DISABLED_FUEL}}
```

Comment:
If you make something into a fuel source, then want to "unmake" it, it fails here.

I don't think the previous code supported it, but it's something that should be looked into.